app/services/reply_scraper.rb

```diff
@@ -0,0 +1,148 @@
+class ReplyScraper < Object
+  BASE_ACCOUNTS = {
+    'alicornucopia'      => 'Alicorn',
+    'pythbox'            => 'Kappa',
+    'lintamande'         => 'lintamande',
+    'marrinikari'        => 'Marri',
+    'peterxy'            => 'Pedro',
+    'peterverse'         => 'Pedro',
+    'curiousdiscoverer'  => 'CuriousDiscoverer',
+    'aestrix'            => 'Aestrix',
+    'unbitwise'          => 'Unbitwise',
+    'erinflight'         => 'ErinFlight',
+    'andaisq'            => 'andaisq',
+    'rockeye-stonetoe'   => 'Rockeye',
+    'rockeye_stonetoe'   => 'Rockeye',
+    'maggie-of-the-owls' => 'MaggieoftheOwls',
+    'maggie_of_the_owls' => 'MaggieoftheOwls', # have both - and _ versions cause Dreamwidth supports both
+    'nemoconsequentiae'  => 'Nemo',
+    'armokgob'           => 'Armok',
+    'timepoof'           => 'Timepoof',
+  }
+
+  attr_reader :errors
+
+  def initialize(reply, errors: ActiveModel::Errors.new(self), console: false)
+    @reply = reply
+    @errors = errors
+    @console_import = console
+  end
+
+  def import(username:, img_url:, img_keyword:, content:, created_at:)
+    @reply.created_at = @reply.updated_at = created_at
+    @reply.content = strip_content(content)
+    @reply.user = set_from_username(username)
+    @reply.icon = set_from_icon(img_url, img_keyword) if img_url
+    post_setup if @reply.is_a? Post
+    Audited.audit_class.as_user(@reply.user) do
+      @reply.save!
+    end
+  end
+
+  private
+
+  def post_setup
+    @reply.last_user_id = @reply.user_id
+    @reply.edited_at = @reply.created_at
```

Comment:
Won't this be nil? The reply hasn't been persisted in the database yet, and Rails doesn't fill this in until it is.

Reply:
...but I set the created_at?

Reply:
...But I set created_at? I guess I didn't set user_id if that's what you mean?

Reply:
Oh, no, I just totally missed the part where you set created_at. Should be fine!